src/main/assisted-installer-controller/assisted_installer_main.go

```diff
@@ -250,8 +250,32 @@ func waitForInstallation(client inventory_client.InventoryClient, log logrus.Fie
 // the inventory client because assisted-service is deployed on the bootstrap node
 // and when the bootstrap node reboots to join the cluster, assisted-service becomes
 // unavailable.
-func waitForInstallationAgentBasedInstaller(kubeClient k8s_client.K8SClient, log logrus.FieldLogger) {
+func waitForInstallationAgentBasedInstaller(kubeClient k8s_client.K8SClient, log logrus.FieldLogger, removeUninitialiedTaint bool) {
 	for {
+		if removeUninitialiedTaint {
+			// After the boostrap node has rebooted, assisted-service becomes unavailable
+			// and nodes can no longer be untainted through
+			// assistedController.WaitAndUpdateNodesStatus(mainContext, &wg, removeUninitializedTaint).
+			// because the call to retrieve hosts (c.ic.GetHosts(ctxReq, log, ignoreStatuses))
+			// will fail immediately.
+			// Here we have an alternative mechanism to untaint nodes when
+			// assisted-service is no longer available.
+			nodes, err := kubeClient.ListNodes()
+			if err != nil {
+				log.WithError(err).Error("Failed to get list of nodes from k8s client")
```

Comment:
Should this return here on error instead of trying to use the nodes value below?

Reply:
I think return should only happen to signal that the cluster has finished installing. On error, nodes.Items will be empty so should be ok. We still want the clusterVersion check to happen below.